cmd/cloudFoundryDeploy.go

```diff
@@ -0,0 +1,855 @@
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/SAP/jenkins-library/pkg/yaml"
+	"github.com/elliotchance/orderedmap"
+	"github.com/pkg/errors"
+	"io"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type cfFileUtil interface {
+	FileExists(string) (bool, error)
+	FileWrite(path string, content []byte, perm os.FileMode) error
+	Getwd() (string, error)
+	Glob(string) ([]string, error)
+	Chmod(string, os.FileMode) error
+}
+
+var _now = time.Now
+var _cfLogin = cfLogin
+var _cfLogout = cfLogout
+var _getManifest = getManifest
+var _substitute = yaml.Substitute
+var fileUtils cfFileUtil = piperutils.Files{}
+
+// for simplify mocking. Maybe we find a more elegant way (mock for CFUtils)
+func cfLogin(c command.ExecRunner, options cloudfoundry.LoginOptions) error {
+	cf := &cloudfoundry.CFUtils{Exec: c}
+	return cf.Login(options)
+}
+
+// for simplify mocking. Maybe we find a more elegant way (mock for CFUtils)
+func cfLogout(c command.ExecRunner) error {
+	cf := &cloudfoundry.CFUtils{Exec: c}
+	return cf.Logout()
+}
+
+const defaultSmokeTestScript = `#!/usr/bin/env bash
+# this is simply testing if the application root returns HTTP STATUS_CODE
+curl -so /dev/null -w '%{response_code}' https://$1 | grep $STATUS_CODE`
+
+func cloudFoundryDeploy(config cloudFoundryDeployOptions, telemetryData *telemetry.CustomData, influxData *cloudFoundryDeployInflux) {
+	// for command execution use Command
+	c := command.Command{}
+	// reroute command output to logging framework
+	c.Stdout(log.Writer())
+	c.Stderr(log.Writer())
+
+	// for http calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+
+	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
+	err := runCloudFoundryDeploy(&config, telemetryData, influxData, &c)
+	if err != nil {
+		log.Entry().WithError(err).Fatalf("step execution failed: %s", err)
+	}
+}
+
+func runCloudFoundryDeploy(config *cloudFoundryDeployOptions, telemetryData *telemetry.CustomData, influxData *cloudFoundryDeployInflux, command command.ExecRunner) error {
+
+	log.Entry().Infof("General parameters: deployTool='%s', deployType='%s', cfApiEndpoint='%s', cfOrg='%s', cfSpace='%s'",
+		config.DeployTool, config.DeployType, config.APIEndpoint, config.Org, config.Space)
+
+	var deployTriggered bool
+	var err error
+
+	if config.DeployTool == "mtaDeployPlugin" {
+		deployTriggered = true
+		err = handleMTADeployment(config, command)
+	} else if config.DeployTool == "cf_native" {
+		deployTriggered = true
+		err = handleCFNativeDeployment(config, command)
+	} else {
+		log.Entry().Warningf("Found unsupported deployTool ('%s'). Skipping deployment. Supported deploy tools: 'mtaDeployPlugin', 'cf_native'", config.DeployTool)
+	}
+
+	if deployTriggered {
+		prepareInflux(err == nil, config, influxData)
+	}
+
+	return err
```

Comment:
Also in groovy we have a warning in the log and nothing else in case of an unsupported deploy tool. I personally think we should fail in such a case, but is is the [same](https://github.com/SAP/jenkins-library/blob/master/vars/cloudFoundryDeploy.groovy#L277) behaviour in groovy --> should be changed in a dedicated PR IMO since here it is above moving groovy to go.